stellar_rust_sdk/src/trades/response.rs

```diff
@@ -0,0 +1,113 @@
+use crate::models::prelude::*;
+use derive_getters::Getters;
+use serde::{Deserialize, Serialize};
+
+/// Represents the response for the 'all trades' query in the Horizon API.
+///
+/// This struct defines the overall structure of the response for an 'all trades' query.
+/// It includes navigational links and embedded results.
+///
+#[derive(Debug, Clone, Serialize, Deserialize, Getters)]
+pub struct AllTradesResponse {
+    #[serde(rename = "_links")]
+    pub links: ResponseLinks,
+    #[serde(rename = "_embedded")]
+    pub embedded: Embedded<TradeResponse>,
+}
+
+impl Response for AllTradesResponse {
+    fn from_json(json: String) -> Result<Self, String> {
+        let response = serde_json::from_str(&json).map_err(|e| e.to_string())?;
+
+        Ok(response)
+    }
+}
+
+/// Represents the navigational links in a single trade response from the Horizon API.
+///
+/// This struct includes various hyperlinks such as links to the trade itself,
+/// base asset, counter asset and operation.
+///
+#[derive(Debug, Deserialize, Serialize, Clone, Getters)]
+pub struct Links {
+    #[serde(rename = "self")]
+    self_link: Link,
+    base: Link,
+    counter: Link,
+    operation: Link,
+}
```

Comment:
Gaat dit goed met de andere preludes?
Neem bijv [AccountResponseLinks](https://github.com/Baseflow/stellar-rust-sdk/blob/58a081ce59fc8e12a995832d0ea848d011f86726/stellar_rust_sdk/src/accounts/response.rs#L75). Door een dedicated naam te gebruiken weten we zeker dat we niet meerdere public structs hebben met dezelfde naam (wat via een prelude voor conflicten kan zorgen).

Reply:
Tot nu toe gaat het goed, maar ik kan me voorstellen dat het inderdaad problemen gaat opleveren; ik ga het aanpassen. Misschien is het goed om te kijken of de andere endpoints ook van deze conventie gebuik maken. :slightly_smiling_face: